src/Sentry.Serilog/SentrySink.cs

```diff
@@ -74,8 +76,18 @@ public void Emit(LogEvent logEvent)
             }
 
             var exception = logEvent.Exception;
-            var formatted = logEvent.RenderMessage(_options.FormatProvider);
             var template = logEvent.MessageTemplate.Text;
+            string formatted;
+            if (_options.TextFormatter is { } formatter)
+            {
+                using var stringWriter = new StringWriter(new StringBuilder());
+                formatter.Format(logEvent, stringWriter);
+                formatted = stringWriter.ToString();
+            }
+            else
+            {
+                formatted = logEvent.RenderMessage(_options.FormatProvider);
+            }
```

Comment:
Is there something like a `PlainTextFormatter` that we can use? So that we could:

```csharp
var formatter = _options.TextFormatter ?? PlainTextFormatter.Instance;
var formatted = ...
```

If not, maybe we can create one?

Finally, if none of those work, can we at least move this part to a separate function like `string Format(LogEvent logEvent)` and replace variable mutations with return statements instead?

Reply:
Another suggestion is to create an internal extension method for `ITextFormatter`:

```csharp
static string Format(this LogEvent logEvent) { /* ... */ }
```

which would use `StringWriter`. The purpose is to move this infrastructural complexity out of this method to make things a bit simpler.

Reply:
I don't think there's a `PlainTextFormatter` available.
We can create a `PlainTextFormatter` or `FallbackTextFormatter` that take a `SentrySerilogOptions` in the constructor since `SentrySerilogOptions.FormatProvider` is also mutable.
Would you want me to make the changes?

Reply:
It sounds like it would be too much work, let's go with just extracting into a separate method